src/classes/CRLP_Batch_Base_NonSkew.cls

```diff
@@ -191,29 +196,41 @@ public abstract class CRLP_Batch_Base_NonSkew extends CRLP_Batch_Base {
                     whereClauses.add(addlFilter);
                 }
 
-                if (householdSettings.npo02__Always_Rollup_to_Primary_Contact__c == false) {
-                    whereClauses.add('(IsPrimary = False OR Opportunity.Account.npe01__SYSTEMIsIndividual__c = False)');
-                } else {
-                    whereClauses.add('IsPrimary = False');
+                if (jobType != CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
+                    if (householdSettings.npo02__Always_Rollup_to_Primary_Contact__c == false) {
+                        whereClauses.add('(IsPrimary = False OR Opportunity.Account.npe01__SYSTEMIsIndividual__c = False)');
+                    } else {
+                        whereClauses.add('IsPrimary = False');
+                    }
                 }
 
                 if (!whereClauses.isEmpty()) {
                     query += ' WHERE ' + String.join(whereClauses, ' AND ');
                 }
 
                 // Note that the ORDER BY here doesn't gaurantee that the records will actually be in this order
-                query += ' ORDER BY ' + this.keyField + ', Opportunity.CreatedDate ASC ';
+                query += ' ORDER BY ' + this.keyField;
 
-                UTIL_Debug.debug(LoggingLevel.WARN, '>> OCR Query =' + query);
+                if (jobType != CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
```

Comment:
Why not sort by Opp.CreatedDate when rolling up AccountSoftCredits?

Reply:
Because the Oppty relationship is different so it was breaking the query. Fixed to make the string added conditional and use `Opportunity.` or `Opportunity__r.` as needed.

Reply:
@bethbrains I see your note about fixing to use a different sort key, but the code above showing using `this.keyField` as the sort. Will that variable contain either `Opportunity.CreatedDate` or `Opportunity__r.CreatedDate`? Also, it's missing the `ASC' sort sequence.